client/dashboard/app/queries/site.ts

```diff
@@ -26,6 +26,23 @@ export const siteBySlugQuery = ( siteSlug: string ) => ( {
 } );
 
 export const siteByIdQuery = ( siteId: number ) => ( {
+	queryKey: [ 'site-by-id', siteId, SITE_FIELDS, SITE_OPTIONS ],
+	queryFn: async () => {
+		try {
+			return await fetchSite( siteId );
+		} catch ( e: any ) /* eslint-disable-line @typescript-eslint/no-explicit-any */ {
+			if ( e.error === 'unknown_blog' ) {
+				throw notFound();
+			}
+			throw e;
+		}
+	},
+	meta: {
+		persist: false,
```

Comment:
Hmm why? With this, every time we refresh a site's overview page, we will be blocked until we finish fetching the staging / production site (because it's not cached).